packages/components/src/dropdown-menu-v2-ariakit/index.tsx

```diff
@@ -181,7 +181,13 @@ const UnconnectedDropdownMenu = (
 		children,
 		shift,
 		modal = true,
-		hideOnEscape = true,
+		hideOnEscape = ( event ) => {
```

Comment:
IMO, since we're just passing down this prop and it's ariakit that is handling the ESCAPE key, I think the ideal fix would have to be done in ariakit directly

Reply:
I'm fine with a fix here for now, but maybe this could benefit more users.

Reply:
@diegohaz  maybe you can share some perspective as to why this is not the case in `ariakit`?

Reply:
I can't remember any discussion about `hideOnEscape` and fullscreen mode in particular, so there's likely no specific reason. It seems it just hasn't been considered yet.

But there's an issue with fullscreen and portal: https://github.com/ariakit/ariakit/issues/865